### PR TITLE
Update outdated packages in package.json

### PR DESCRIPTION
**Pull request by konti-aditi**
Date: 09-10-2024

Developer Name: Aditi Shinde

---

## Issue Ticket Number
#454 

## Description
This Updates the following packages from the package.json

• async-storage     • netinfo     • masked-view     • axios     • prettier     • app-auth     • tab-view     • image-picker     • keychain   • buttons-group    • safe-area-context     • react-native-svg     • react-redux     • align-deps      • jest-native     • jest     • react     • react-native      • eslint-parser     • eslint

Additionally updated DatePicker.tsx for compatibility with the updated dependencies (Refs: #462 )

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

</details>

[](https://github.com/user-attachments/assets/b3f51606-3ed4-4eb7-aae5-fc24b6da8549)

## Test Coverage
<details>
<summary>Screenshot 1</summary>
NA

</details>

### CODE REVIEW

File: android/build.gradle
Comment (ashifkhn) [inline android/build.gradle:1]: Why have you changed minSdkVersion?

File: android/gradle.properties
Comment (ashifkhn) [inline android/gradle.properties:1]: Why was this change needed?

File: src/components/CreateGoalForm/DatePicker.tsx
Comment (ashifkhn) [inline src/components/CreateGoalForm/DatePicker.tsx:1]: As per our discussion, we decided to remove this code in a separate pr.

**Review by ashifkhn — CHANGES_REQUESTED** [review]
Left few comments;

File: android/build.gradle
Comment (Ayushsanjdev) [reply android/build.gradle:1]: Correct, This will lead to having issues on publishing it to playstore, should always use latest. @Aditys018 fyi

**Review by Ayushsanjdev — CHANGES_REQUESTED** [review]

File: src/components/CreateGoalForm/DatePicker.tsx [hunk 1]
```diff
@@ -1,58 +1,58 @@
-import React, { useState } from 'react';
-import { SafeAreaView, StyleSheet, View } from 'react-native';
-import DatePicker from 'react-native-datepicker';
+// import React, { useState } from 'react';
```
Comment (shreya-mishra) [inline src/components/CreateGoalForm/DatePicker.tsx]: why are we commenting this file , if we are not using it please delete it

File: package.json [hunk 2]
```diff
@@ -17,53 +17,52 @@
   },
   "dependencies": {
     "@babel/helper-hoist-variables": "^7.24.7",
-    "@react-native-async-storage/async-storage": "^1.15.16",
-    "@react-native-community/netinfo": "^5.9.9",
+    "@react-native-async-storage/async-storage": "^1.24.0",
+    "@react-native-community/netinfo": "^5.9.10",
     "@react-native-community/slider": "^4.4.3",
-    "@react-native-masked-view/masked-view": "^0.2.6",
+    "@react-native-masked-view/masked-view": "^0.2.9",
```
Comment (shreya-mishra) [inline package.json:23]: can we pinned these versions as per our discussion ?

**Review by ashifkhn — DISMISSED** [review]

**Review by pankajjs — APPROVED** [review]

**Comment by vinit717** [comment]
CI/CD build is failing and also add screenshot of working app build

**Review by vinit717 — CHANGES_REQUESTED** [review]
CI build is failing and also add screenshot of working app build

**Comment by konti-aditi** [comment]
@vinit717  CI Build is successful. 
Recording of the working app here : 

https://github.com/user-attachments/assets/33761aef-699b-44b5-9811-f3120e81344a

File: package.json [hunk 3]
```diff
@@ -77,18 +77,17 @@
...
     "@react-native/typescript-config": "0.74.84",
-    "@rnx-kit/align-deps": "^2.2.2",
-    "@testing-library/jest-native": "^4.0.4",
+    "@rnx-kit/align-deps": "^2.5.5",
+    "@testing-library/jest-native": "^4.0.13",
     "@testing-library/react-native": "^12.3.0",
-    "@types/jest": "^26.0.23",
-    "@types/react": "^18.2.6",
-    "@types/react-native": "^0.66.4",
-    "@types/react-native-datepicker": "^1.7.1",
```
Comment (vinit717) [inline package.json:86]: Are we not using it? `@types/react-native-datepicker`

**Comment by konti-aditi** [comment]
Resolved date-picker comment by @vinit717 


https://github.com/user-attachments/assets/ae222bc6-d020-4ed1-bb9e-6c1dea56568a

**Review by ashifkhn — APPROVED** [review]

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**